### PR TITLE
BrAPI trait duplicate errors more specific. Added info messages to st…

### PR DESCRIPTION
**Pull request by ctucker3**
Displays a message when a duplicate trait is attempted to be imported through BrAPI. If there is more than one duplicate trait, the last trait that is a duplicate will be displayed in the message. 

Additional info: If many traits are selected, traits that are not duplicated will be inserted, but traits that are duplicates will be skipped.

### CODE REVIEW

**Comment by ctucker3** [comment]
[FIE-57 - Make duplicate trait error more specific](https://trello.com/c/TVRi9syA/39-fie-57-make-duplicate-trait-error-more-specific)

File: app/src/main/java/com/fieldbook/tracker/brapi/BrapiTraitActivity.java [hunk 1]
```diff
@@ -280,14 +280,14 @@ public String saveTraits() {
...
+            return secondaryMessage != "" ? secondaryMessage : getResources().getString(R.string.brapi_error_saving_all_traits);
         }
         else if (successfulSaves < totalTraits) {
-            return "Error saving some traits. Some traits were not saved";
+            return secondaryMessage != "" ? secondaryMessage : getResources().getString(R.string.brapi_error_saving_some_traits);
         }
 
         // Check if we had a secondary message to show
-        return secondaryMessage != "" ? secondaryMessage : "Selected traits saved successfully.";
+        return secondaryMessage != "" ? secondaryMessage : getResources().getString(R.string.brapi_traits_saved_success);
```
Comment (nickpalladino) [inline app/src/main/java/com/fieldbook/tracker/brapi/BrapiTraitActivity.java]: Discussed adding a comment here to make it clear that although it currently doesn't have any other secondary messages, it is written this way for possible future messages that may be added.

File: app/src/main/java/com/fieldbook/tracker/brapi/BrapiTraitActivity.java [hunk 2]
```diff
@@ -258,7 +258,7 @@ public String saveTraits() {
 
             // Check if the trait already exists
             if (ConfigActivity.dt.hasTrait(trait.getTrait())) {
-                secondaryMessage = "Trait already exists: " + trait.getTrait();
+                secondaryMessage = getResources().getString(R.string.brapi_trait_already_exists, trait.getTrait());
```
Comment (nickpalladino) [inline app/src/main/java/com/fieldbook/tracker/brapi/BrapiTraitActivity.java]: Discussed rationale for only showing last duplicate trait which was for a good balance of providing more information while not requiring an overly involved implementation. In the future, may add information on all of the traits in a dialog.

**Comment by ctucker3** [comment]
Changes have been implemented.

**Pull request merged**